lib/config/types.ts

```diff
@@ -95,6 +95,7 @@ export interface GlobalOnlyConfig {
   privateKeyPathOld?: string;
   redisUrl?: string;
   repositories?: RenovateRepository[];
+  endpoint?: string;
```

Comment:
I think can be removed since `endpoint` is also used non-globally in `ExtractConfig` can be removed in another PR which close #15800 ?

Reply:
you need to fix all places in one PR for this

Reply:
So the one PR will contain all steps in one place, ie.
- Add new type to `GlobalConfig`
- change `defaultRegistryUrls` type
- use global config to implement logic for #15800

Reply:
no, only all changes required by removing endpoint from normal renovate config. checkout previous PR which moved such options

Reply:
We can't remove `endpoint` from `RenovateConfig` only from `ExtractConfig`

Reply:
why not?

Reply:
Because we need `endpoint` to be in the config as we use repoConfig which is of type `RenovateConfig` to set global config
https://github.com/renovatebot/renovate/blob/6257604ec8e8d5880831023a99bc3b27a98246d0/lib/workers/repository/index.ts#L22-L29